pkg/utils/jsonutils_test.go

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package utils
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/heketi/tests"
+)
+
+type testDest struct {
+	Foo string `json:"foo"`
+	Bar int    `json:"bar"`
+}
+
+type dummyReader struct {
+	emsg string
+}
+
+type dummyCloser struct {
+	io.Reader
+}
+
+func (d *dummyReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New(d.emsg)
+}
+
+func (c dummyCloser) Close() error { return nil }
+
+func TestGetJsonFromRequestOk(t *testing.T) {
+	dest := &testDest{}
+	req, err := http.NewRequest(
+		http.MethodPut,
+		"/foo/bar",
+		bytes.NewBufferString(`{"foo": "MyTest", "bar": 88}`))
+	tests.Assert(t, err == nil, "err is nil", err)
+	err = GetJsonFromRequest(req, dest)
+	tests.Assert(t, err == nil, "err is nil", err)
+	tests.Assert(t, dest.Foo == "MyTest",
+		"failed check:", dest.Foo, "==", "MyTest")
+	tests.Assert(t, dest.Bar == 88,
+		"failed check", dest.Bar, "==", 88)
+}
+
+func TestGetJsonFromRequestBadJson(t *testing.T) {
+	dest := &testDest{}
+	req, err := http.NewRequest(
+		http.MethodPut,
+		"/foo/bar",
+		bytes.NewBufferString(`{"foo": `))
+	tests.Assert(t, err == nil, "err is nil", err)
```

Comment:
Er, now I read it again, isn't the logic reversed: The Assert will trigger when the err == nil condition is *not* met, hence, the error message should not read "err is nil" but rather reflect the fact that an error occurred. like
```
tests.Assert(t, err == nil, "Error calling http.NewRequest:", err)
```
I think several messages below have this same issue...